src/Form/DataTransformer/ModelToIdPropertyTransformer.php

```diff
@@ -130,7 +131,12 @@ public function transform($entityOrCollection)
         }
 
         foreach ($collection as $entity) {
-            $id = current($this->modelManager->getIdentifierValues($entity));
+            $ids = $this->modelManager->getIdentifierValues($entity);
+            if (count($ids) > 1) {
```

Comment:
Maybe it is better to check if it is an array? or it is always an array?

Reply:
https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/3.x/src/Model/ModelManager.php#L332, always array

Reply:
Always an array but current() returns the first item, $id was in fact whatever type.
The tests considers it's an integer, but if you use a UUID, it's not.

To be correct the call to getIdentifierValues should be replaced by getNormalizedIdentifier but:
 - previous fixes to allow composite keys use the same kind of code [here](https://github.com/sonata-project/SonataAdminBundle/blob/d6ffaef34cf6e0b08a12478478d5105eec592b50/src/Form/DataTransformer/ModelsToArrayTransformer.php#L130) or [here](https://github.com/sonata-project/SonataAdminBundle/blob/d6ffaef34cf6e0b08a12478478d5105eec592b50/src/Form/ChoiceList/ModelChoiceLoader.php#L114)
 - this leads to breaking types used in the tests suite

Reply:
Can we fix the same way we did in other parts of sonata? IMO it would be more consistent that way

Reply:
But IMO, what makes sense is to use getNormalizedIdentifier as you said. wdyt on replacing those getIdentifierValues you found with the implode by the getNormalizedIdentifier?